sparse_strips/vello_cpu/src/fine/mod.rs

```diff
@@ -507,20 +518,324 @@ pub(crate) mod strip {
+    }
+
+    #[inline(always)]
+    fn extract_color(color: &PremulColor) -> [Self; COLOR_COMPONENTS] {
+        color.as_premul_rgba8().to_u8_array()
+    }
+
+    #[inline(always)]
+    fn from_normalized_u8(num: u8) -> Self {
+        num
+    }
+
+    #[inline(always)]
+    fn from_u8(num: u8) -> Self {
+        num
+    }
+
+    #[inline(always)]
+    fn to_f32_normalized(self) -> f32 {
+        self as f32 / 255.0
+    }
+
+    #[inline(always)]
+    fn from_normalized_f32(num: f32) -> Self {
+        (num * 255.0 + 0.5) as Self
+    }
+
+    #[inline(always)]
+    fn to_rgba8(_in: &[Self]) -> [u8; COLOR_COMPONENTS] {
+        [_in[0], _in[1], _in[2], _in[3]]
+    }
+
+    #[inline(always)]
+    fn from_rgba8(_in: &[u8]) -> [Self; COLOR_COMPONENTS] {
+        [_in[0], _in[1], _in[2], _in[3]]
+    }
+
+    #[inline(always)]
+    fn from_rgbaf32(_in: &[f32]) -> [Self; COLOR_COMPONENTS] {
+        let r = |val: f32| (val * 255.0 + 0.5) as Self;
+        [r(_in[0]), r(_in[1]), r(_in[2]), r(_in[3])]
+    }
+
+    #[inline(always)]
+    fn inv(self) -> Self {
+        Self::ONE - self
+    }
+
+    #[inline(always)]
+    fn widen(self) -> Self::Widened {
+        self as u16
+    }
+}
+
+impl FineType for f32 {
+    type Widened = Self;
+    const ZERO: Self = 0.0;
+    const MID: Self = 0.5;
+    const ONE: Self = 1.0;
+
+    #[inline(always)]
+    fn min(self, other: Self) -> Self {
+        Self::min(self, other)
+    }
+
+    #[inline(always)]
+    fn max(self, other: Self) -> Self {
+        Self::max(self, other)
+    }
+
+    fn widened_mul_div(self, other: Self, other2: Self) -> Self {
+        (self * other) / other2
+    }
+
+    #[inline(always)]
+    fn normalized_mul(self, other: Self) -> Self {
+        self * other
+    }
+
+    #[inline(always)]
+    fn extract_color(color: &PremulColor) -> [Self; COLOR_COMPONENTS] {
+        color.as_premul_f32().components
+    }
+
+    #[inline(always)]
+    fn from_normalized_u8(num: u8) -> Self {
+        num as Self / 255.0
+    }
+
+    #[inline(always)]
+    fn from_u8(num: u8) -> Self {
+        num as Self
+    }
+
+    #[inline(always)]
+    fn to_f32_normalized(self) -> f32 {
+        self
+    }
+
+    #[inline(always)]
+    fn from_normalized_f32(num: f32) -> Self {
+        num
+    }
+
+    #[inline(always)]
+    fn to_rgba8(_in: &[Self]) -> [u8; COLOR_COMPONENTS] {
+        let mut out = [0; COLOR_COMPONENTS];
+
+        for i in 0..COLOR_COMPONENTS {
+            out[i] = (_in[i] * 255.0 + 0.5) as u8;
+        }
+
+        out
+    }
+
+    #[inline(always)]
+    fn from_rgba8(_in: &[u8]) -> [Self; COLOR_COMPONENTS] {
+        let c = |val: u8| val as Self / 255.0;
+        [c(_in[0]), c(_in[1]), c(_in[2]), c(_in[3])]
+    }
+
+    #[inline(always)]
+    fn from_rgbaf32(_in: &[f32]) -> [Self; COLOR_COMPONENTS] {
+        [_in[0], _in[1], _in[2], _in[3]]
+    }
+
+    #[inline(always)]
+    fn inv(self) -> Self {
+        1.0 - self
+    }
```

Comment:
These can become default and be deleted here.

```suggestion
    #[inline(always)]
    fn to_u8(self) -> u8 {
        ((self * 255.) + 0.5) as u8
    }
```